synapse/http/servlet.py

```diff
@@ -70,12 +70,20 @@ def parse_integer(request: Request, name: str, *, required: Literal[True]) -> in
 
 @overload
 def parse_integer(
-    request: Request, name: str, default: Optional[int] = None, required: bool = False
-) -> Optional[int]: ...
+    request: Request,
+    name: str,
+    default: Optional[int] = None,
+    required: bool = False,
+    negative: bool = False,
+) -> int: ...
```

Comment:
```suggestion
) -> Optional[int]: ...
```
Or why this was changed?

Reply:
Originally it was changed to satisfy mypy.
But you are right; there are cases where these would not be correct.
I added (fixed) those overloads to satisfy those signatures, and make mypy happy.
Thanks for the hint!